golang/cosmos/app/app.go

```diff
@@ -965,9 +977,14 @@ func (app *GaiaApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci
 		panic(err)
 	}
 
+	ensureBlockMaxBytes(app, ctx)
```

Comment:
If we're starting a new chain, rather than upgrading a running chain, then we ought to fail for a too large `Block.MaxBytes` rather than silently capping it, which will confound future maintainers. The lower value should be explicit in the genesis file. I suggest writing a function `ValidateConsensusParams` and refer to `types/ConsensusParams.ValidateBasic` in the consensus library.